librocksdb_sys/crocksdb/c.cc

```diff
@@ -1489,6 +1483,17 @@ void crocksdb_options_set_block_based_table_factory(
   }
 }
 
+size_t crocksdb_options_get_block_cache_usage(crocksdb_options_t *opt) {
+  if (opt) {
+    if (opt->rep.table_factory != nullptr) {
+      void* block_based_table_opt = opt->rep.table_factory->GetOptions();
+      if (block_based_table_opt) {
+        return static_cast<BlockBasedTableOptions*>(block_based_table_opt)->block_cache->GetUsage();
```

Comment:
do we need to check the table factory is BlockBasedTableFactory first? If not BlockBasedTableFactory, this function may panic.

We can use BlockBasedTableFactory function name() == "BlockBasedTable"  to check it.